### ISSUE TITLE
cabal got less smart about finding ghc-pkg

### ISSUE THREAD

**Issue by rwbarton** [haskell/cabal#3150]
Cabal used to be better about figuring out what ghc-pkg to run based on the compiler command name.
```
rwbarton@morphism:/tmp/haskell-obdd$ cabal-newest --version
cabal-install version 1.22.4.0
using version 1.22.2.0 of the Cabal library 
rwbarton@morphism:/tmp/haskell-obdd$ cabal-newest configure -w ghc-7.10.1
Resolving dependencies...
Configuring obdd-0.3.3...
cabal-newest: At least the following dependencies are missing:
mtl -any, random -any
```
But since Cabal 1.23.1.0 (or possibly cabal-install 1.23.0.0) it no longer figures this out automatically in this case.
```
rwbarton@morphism:/tmp/haskell-obdd$ cabal-super --version
cabal-install version 1.23.0.0
compiled using version 1.23.1.0 of the Cabal library 
rwbarton@morphism:/tmp/haskell-obdd$ cabal-super configure -w ghc-7.10.1
cabal-super: Version mismatch between ghc and ghc-pkg:
/home/rwbarton/bin/ghc-7.10.1 is version 7.10.1 /home/rwbarton/bin/ghc-pkg is
version 7.8.4
```

**Comment by rwbarton** [comment]
It would be nice if `cabal-super configure -w i386-unknown-linux-gnu-ghc` worked, too, finding `i386-unknown-linux-gnu-ghc-pkg`. But that's a feature request, not a regression.

**Comment by ezyang** [comment]
I touched this code recently.  Would you mind checking if a218453869579720821e0a40394be11e9155dac1 was the breaking commit?

It would also be helpful to know how your symlink structure is set up.

**Comment by rwbarton** [comment]
`/home/rwbarton/bin/ghc-7.10.1` is not a symlink. `/home/rwbarton/bin/ghc-pkg` is a symlink, to `ghc-pkg-7.8.4`.

**Comment by ezyang** [comment]
So, `/home/rwbarton/bin/` is in your PATH, and there exists a `/home/rwbarton/bin/ghc-pkg-7.10.1` (also not symlink)?

**Comment by rwbarton** [comment]
Correct.

**Comment by rwbarton** [comment]
Yes, it worked in a218453^ but broke in a218453.

**Comment by ezyang** [comment]
I see the braino.

**Comment by ezyang** [comment]
I have a fix but I think it will take a bit longer to construct a test case

**Comment by 23Skidoo** [comment]
Fixed by #3153.
